sys/posix/include/semaphore.h

```diff
@@ -6,117 +6,170 @@
  * directory for more details.
  */
 
-#ifndef _SEMAPHORE_H
-#define _SEMAPHORE_H    1
+/**
+ * @defgroup    posix_semaphore POSIX semaphores
+ * @ingroup     posix
+ * @{
+ * @file
+ * @brief   Semaphores
+ * @see     <a href="http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/semaphore.h.html">
+ *              The Open Group Base Specifications Issue 7, <semaphore.h>
+ *          </a>
+ * @author  Christian Mehlis <mehlis@inf.fu-berlin.de>
+ * @author  Martine Lenders <mlenders@inf.fu-berlin.de>
+ * @author  René Kijewski <kijewski@inf.fu-berlin.de>
+ */
 
+#ifndef _POSIX_SEMAPHORE_H
```

Comment:
(better to comment in the PR than in the commit)
remove the leading underscore and add a trailing if you like